app/misc/repo.ts

```diff
@@ -258,11 +259,85 @@ function openRepository() {
     }
   }
 
+// works as a monitor for any change to the reference list
+function refreshList() {
+  Git.Repository.open(repoFullPath)
+    .then(function (repo) {
+      repo.getCurrentBranch()
+        .then(function () {
+          // grab the list
+          return repo.getReferences(Git.Reference.TYPE.LISTALL);
+        })
+        .then(function (refList) {
+          // sort refList alphabetically to get uniform order of the list
+          refList.sort();
+
+          // monitor any changes to reference list
+          if (lastRefList.length === refList.length && lastRefList.every(function(value, index) { return value.name() === refList[index].name()})) {
```

Comment:
This is a naive comparison of the two list of references.
Possibly, there could be a better solution than mine, but it passed all my test cases and gets the job done.